client/state/data-layer/utils.js

```diff
@@ -0,0 +1,17 @@
+/**
+ * External dependencies
+ */
+import {
+	isArray,
+	mergeWith,
+} from 'lodash';
+
+const concatHandlers = ( left, right ) =>
+	isArray( left )
+		? left.concat( right )
+		: undefined;
+
+export const mergeHandlers = ( ...handlers ) =>
+	handlers.length > 1
+		? mergeWith( {}, ...handlers, concatHandlers )
+		: handlers.shift();
```

Comment:
- What's the use case for passing a single handler to this function?
- Would it be more reasonable to accept a single array argument instead of the spread arguments? What if we want to add an options argument in the future?